pkg/deployment/agent_test.go

```diff
@@ -59,6 +59,18 @@ func TestGetDaemonSetDeployment(t *testing.T) {
 
 	ds := agent.Get()
 	assert.NotNil(t, ds)
+}
+
+func TestDaemonSetAgentAnnotations(t *testing.T) {
+	jaeger := v1alpha1.NewJaeger("TestDaemonSetAgentAnnotations")
+	jaeger.Spec.Agent.Strategy = "daemonset"
+	jaeger.Spec.Agent.Annotations = map[string]string{
+		"hello": "world",
+	}
+
+	agent := NewAgent(jaeger)
+	dep := agent.Get()
 
-	assert.Equal(t, "false", ds.Spec.Template.Annotations["sidecar.istio.io/inject"])
+	assert.Equal(t, "false", dep.Spec.Template.ObjectMeta.Annotations["sidecar.istio.io/inject"])
```

Comment:
Could you also add a test where you override an annotation that we set implicitly, like `prometheus.io/scrape`? The code in this PR makes me believe that a user is able to set this to `false`, so, having a test that explicitly tests this behavior would also serve as documentation.